scipy/optimize/_dual_annealing.py

```diff
@@ -402,6 +410,10 @@ def __init__(self, search_bounds, func_wrapper, **kwargs):
                 'maxiter': ls_max_iter,
             }
             self.kwargs['bounds'] = list(zip(self.lower, self.upper))
+        if 'jac' in self.kwargs:
+            # Make sure jac wrapper is used so that args are passed.
+            func_wrapper.jac = self.kwargs['jac']
```

Comment:
I'm not sure this alteration will work. From the `minimize` [documentation for jac](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize), `jac` can be one of `{callable, ‘2-point’, ‘3-point’, ‘cs’, bool}`. So the `minimizer_kwargs['jac']` is not necessarily a callable.

Reply:
Good point! Thanks @andyfaff, then, we would need this only when ```jac``` is a callable, and if it is anything else, the ```func_wrapper.jac``` is not used and the ```jac``` option is passed as-is to the local minimizer.